node/impl/full/state.go

```diff
@@ -422,6 +422,29 @@ func (a *StateAPI) StateReadState(ctx context.Context, actor address.Address, ts
 	}, nil
 }
 
+func (a *StateAPI) StateDecodeParams(ctx context.Context, msgCid cid.Cid, tsk types.TipSetKey) (interface{}, error) {
```

Comment:
Maybe we should just have this take a target actor, method ID and params bytes? That makes this quite a bit more flexible

Reply:
donesies